sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -1681,8 +1681,9 @@ object GpuOverrides extends Logging {
             .withPsNote(TypeEnum.STRING, "A limited number of formats are supported"),
             TypeSig.STRING)),
       (a, conf, p, r) => new UnixTimeExprMeta[DateFormatClass](a, conf, p, r) {
+        override def isTimeZoneSupported = true
```

Comment:
Why do we need this?

Reply:
This is due to the fact that DateFormatClass is not considered a TimeZoneAwareExpression, but requires support for non-UTC timezones. It's the last check for timezone requirement in Expressions. By default, this flag is `false`.

Reply:
@ttnghia FYI. Some original [notes](https://github.com/NVIDIA/spark-rapids/pull/9719#issue-1994351226) around this.

Reply:
It seems not a good practice to set some value as `false` by default then override `true` like this. Can we compute that value based on the input in the object constructor?